### DISCUSSION TITLE
how to configure the IP/network settings?

### DISCUSSION THREAD

**Discussion by alexisdal** [Tiryoh/docker-ros-desktop-vnc#26] [General]
first of all, thanks for making a simple ROS container. I'm working on windows and i struggle to understand the getting started procedure on Windows. A docker container feels just great, in particular the ability to jump from a ROS version to another.

It's probably my ignorance, but how should I configure my station to have a container that connects to my network environment so that the container can later connect to existing ros robots already present on my network? I typically I want to observe ros topics in rviz from the real robots.

right now it says
```
# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/sit 0.0.0.0 brd 0.0.0.0
8: eth0@if9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
    link/ether 02:42:ac:11:00:02 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 172.17.0.2/16 brd 172.17.255.255 scope global eth0
       valid_lft forever preferred_lft forever
#
```

and i need it to ask for a dhcp lease to my router so that it gets an ip 192.168.1.x ? 

Any hints on how to get started ?

**Comment by Tiryoh** [comment]
@alexisdal Thanks for using GitHub Discussions!
I haven't used Docker on Windows recently, so I'll check it out.

**Reply by alexisdal** [reply]
@Tiryoh thanks.

for the record. by editing the .bashrc and changing the ROS_* environement variable inside the container like below:

```
#export ROS_IP=127.0.0.1
export ROS_IP=192.168.1.200
export ROS_MASTER_URI=http://$ROS_IP:11311
```
I can list ros topics from my robot (192.168.1.200).

However, when I actually try to read messages (rostopic echo or rostopic hz) then it does not work. I suspect this is because when I try to this the container is sending a request to the robot to push its messages to the container, and that's where it fails.

I wouldn't be surprised I need to create a macvlan on the docker container configuration
https://docs.docker.com/network/macvlan/
But i just don't know how to do it. I simply do not want to create a VirtualBox, fire a pre-made ubuntu desktop 20 image, install the entire ros-desktop environment => and then do the network configuration in virtualbox (this I know how to do it... in docker no)

maybe that can help ?

**Reply by Tiryoh** [reply]
@alexisdal Thanks, detailed information would be very helpful to investigate Docker on Windows. I'll check it out today.

By the way, do you use WSL2 for backend?
https://docs.docker.com/docker-for-windows/wsl/

**Reply by Tiryoh** [reply]
I haven't been able to find a solution yet, but I found some important information that I would like to share.

According to the following page, the `macvlan` option works only on Linux.

> The macvlan networking driver only works on Linux hosts, and is not supported on Docker Desktop for Mac, Docker Desktop for Windows, or Docker EE for Windows Server.

https://docs.docker.com/network/network-tutorial-macvlan/

It seems like that the transparent network driver is required on windows.

https://forums.docker.com/t/windows-10-place-container-directly-on-local-network/30247

https://docs.microsoft.com/en-us/virtualization/windowscontainers/container-networking/network-drivers-topologies
